Build fails on musl due to missing type

Original report: https://bugs.gentoo.org/924165

When using the [musl C library](https://www.musl-libc.org/), semigroups fails to build with the error,

```
src/to_gap.hpp: In member function 'OpaqueBag* gapbind14::to_gap<libsemigroups::PBR>::operator()(libsemigroups::PBR) const':
src/to_gap.hpp:469:12: error: 'u_int32_t' was not declared in this scope; did you mean 'uint32_t'?
  469 |       for (u_int32_t i = 0; i < 2 * x.degree(); i++) {
      |            ^~~~~~~~~
      |            uint32_t
```

From what I gather, the `u_int32_t` alias is a GNUism, defined in `sys/types.h` shipped by glibc, but not guaranteed to exist anywhere else. The musl libc, for example, does not typedef it unless `#if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)`.

Hopefully this is as simple as deleting the first underscore to obtain the ISO type `uint32_t`.

--- a/src/to_gap.hpp
+++ b/src/to_gap.hpp
@@ -466,7 +466,7 @@ namespace gapbind14 {
       // empty
       SET_LEN_PLIST(result, 2 * x.degree() + 1);
       SET_ELM_PLIST(result, 1, INTOBJ_INT(x.degree()));
-      for (u_int32_t i = 0; i < 2 * x.degree(); i++) {
+      for (uint32_t i = 0; i < 2 * x.degree(); i++) {
         Obj next = SUM(to_gap<std::vector<uint32_t>>()(x[i]), INTOBJ_INT(1));
         SET_ELM_PLIST(result, i + 2, next);
         CHANGED_BAG(result);

